### PR TITLE
Set up DiscoveryIndexer

### PR DESCRIPTION
**Pull request by michalsidzej**
Resolves L2B-2154

This PR introduces new .env variables:
```ETHEREUM_RPC_URL=
ETHERSCAN_API_KEY=

#-----OPTIONAL-----

CLOCK_INTERVAL_MS=
```

Using `@l2beat/discovery` required `ethers` degradation to `5.7.2`. 

Right now Discovery takes ~20 seconds locally, should be a bit faster on the server.

### CODE REVIEW

File: packages/backend/src/indexers/DiscoveryIndexer.test.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    it('should run discovery on the latest block', async () => {
+      const config = mockConfig()
+      const discoveryEngine = mockObject<DiscoveryEngine>({
+        discover: async () => [],
+      })
+      const discoveryRepository = mockObject<DiscoveryRepository>({
+        addOrUpdate: async () => true,
+      })
+
+      const disocoveryIndexer = new DiscoveryIndexer(
```
Comment (sdlyy) [inline packages/backend/src/indexers/DiscoveryIndexer.test.ts:23]: just some spelling typos
`disocoveryIndexer` ➡️ `discoveryIndexer`

File: packages/backend/src/indexers/DiscoveryIndexer.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      )
+
+      expect(await disocoveryIndexer.update(0, 1)).toEqual(1)
+      expect(discoveryEngine.discover).toHaveBeenCalledTimes(1)
+      expect(discoveryEngine.discover).toHaveBeenNthCalledWith(1, config, 1)
+      expect(discoveryRepository.addOrUpdate).toHaveBeenNthCalledWith(1, {
+        discoveryOutput: {
+          version: 2,
+          name: 'test',
+          configHash: configHash,
```
Comment (sdlyy) [inline packages/backend/src/indexers/DiscoveryIndexer.test.ts:44]:
```suggestion
          configHash: config.hash,
```

Can we strip config hash out of module scope and either declare it per case or use it by accessing it by mock config object? Just a suite's dependency purity

**Review by sdlyy — APPROVED** [review]

**Comment by sdlyy** [comment]
Also sanity check:

Please take a look at `EthereumDiscoveryModule.ts:50`. We've hard-coded the block we should start from there. We should either make it configurable or, I don't know 😅

File: packages/backend/src/indexers/DiscoveryIndexer.test.ts (hunk 1 above)
Comment (michalsidzej) [reply packages/backend/src/indexers/DiscoveryIndexer.test.ts:23]: 6a3997a

File: packages/libs/src/ChainId.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  10: 'optimism',
+  137: 'polygon-pos',
+  56: 'bsc',
+  43114: 'avalanche',
+  42220: 'celo',
+  59144: 'linea',
+  8453: 'base',
+  1101: 'polygon-zkevm',
+  100: 'gnosis',
+  '-1': 'native',
```
[the remark this answers is not in the document]
Comment (michalsidzej) [reply packages/libs/src/ChainId.ts]: Removed: d4188f3

**Comment by michalsidzej** [comment]
> Also sanity check:
> 
> Please take a look at `EthereumDiscoveryModule.ts:50`. We've hard-coded the block we should start from there. We should either make it configurable or, I don't know 😅

e43542e

File: packages/backend/src/indexers/DiscoveryIndexer.test.ts (hunk 2 above)
Comment (michalsidzej) [reply packages/backend/src/indexers/DiscoveryIndexer.test.ts]: f2e3641

**Pull request merged**